src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -157,7 +159,7 @@ public TTagHelper CreateTagHelper<TTagHelper>() where TTagHelper : ITagHelper, n
         /// </remarks>
         public void StartWritingScope()
         {
-            StartWritingScope(new StringWriter());
+            StartWritingScope(new StringCollectionTextWriter(Encoding.UTF8));
```

Comment:
Alternatively use the `Encoding` from the current `TextWriter` - `StartWritingScope(new StringCollectionTextWriter(Output.Encoding));`